lib/go-tc/deliveryservices.go

```diff
@@ -390,6 +392,22 @@ func (ds *DeliveryServiceNullable) validateTypeFields(tx *sql.Tx) error {
 		"orgServerFqdn": validation.Validate(ds.OrgServerFQDN,
 			validation.By(requiredIfMatchesTypeName([]string{DNSRegexType, HTTPRegexType}, typeName)),
 			validation.NewStringRule(validateOrgServerFQDN, "must start with http:// or https:// and be followed by a valid hostname with an optional port (no trailing slash)")),
+		"rangeSliceBlockSize": validation.Validate(ds,
+			validation.By(func(dsi interface{}) error {
+				ds := dsi.(*DeliveryServiceNullable)
+				if ds.RangeRequestHandling != nil {
+					if *ds.RangeRequestHandling == 3 {
+						if ds.RangeSliceBlockSize == nil {
+							return errors.New("rangeSliceBlockSize must be set if the rangeRequestHandling is set to 3 (Use the Slice Plugin)")
+						}
+					} else {
+						if ds.RangeSliceBlockSize != nil {
+							return errors.New("rangeSliceBlockSize can only be set if the rangeRequestHandling is set to 3 (Use the Slice Plugin)")
+						}
+					}
+				}
+				return nil
+			})),
```

Comment:
Since you removed value checking from the API validations, giving it a value for rangeSliceBlockSize that isn't in the acceptable range throws a database error, which gets reported to the user as an Internal Server Error. I think a Bad Request response with appropriate alerts is a better UX.

There's two ways to do this, I think. You can add a case to [`api.ParseDBError`](https://godoc.org/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api#ParseDBError) which _should_ cause it to Just Work :tm: . Or, you could just add back in the constraints here. I don't think having the constraints in both the API and the database is a bad thing, so I'm not really particular what approach you take.

Reply:
ahhh great catch!